src/compiler/commandLineParser.ts

```diff
@@ -564,6 +564,8 @@ namespace ts {
         {
             name: "strict",
             type: "boolean",
+            // Though this affects semantic diagnostics, affectsSemanticDiagnostics is not set here
+            // The value of each strictFlag depends on own strictFlag value or this and never accessed directly.

```

Comment:
I kind of wonder whether it's just worth it to say this always has `affectsSemanticDiagnostics` set to `true` since it's unlikely that you'd toggle this but have no difference in effective strict options.

Reply:
I think its better that we dont update unnecessarily so prefer to keep it like this. But not stuck on it so open to changing that if you feel strongly about it.

Reply:
What’s the actual consequence of doing it like this? Like if someone had every individual strict flag enabled, and then toggled `strict` itself, we would be able to correctly identify that it had no effect?

Reply:
correct.